BenchmarkSample/benchmark/src/androidTest/java/com/example/benchmark/RecyclerViewBenchmark.kt

```diff
@@ -0,0 +1,85 @@
+package com.example.benchmark
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.benchmark.BenchmarkRule
+import androidx.lifecycle.Lifecycle
+import androidx.test.annotation.UiThreadTest
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.filters.LargeTest
+import com.example.benchmark.ui.MainActivity
+import org.junit.After
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/**
+ * RecyclerView benchmark - scrolls a RecyclerView, and measures the time taken to reveal each item
+ *
+ * You can use this general approach to benchmark the performance of RecyclerView. Some things to be aware of:
+ *
+ * - Benchmark one ItemView type at a time. If you have for example section headers, or other types of item variation,
+ *      it's recommended to use fake adapter data with just a single type of item at a time.
+ * - If you want to benchmark TextView performance, use randomized text. Reusing words between items (such as in this
+ *      simple test) will artificially perform better due to unrealistic caching.
+ * - You won't see the effects of RecyclerView prefetching, or Async text layout with this simple approach. We'll be
+ *      adding more complex RecyclerView examples as time goes on.
+ *
+ * Note that this benchmark measures the sum of multiple potentially expensive stages of displaying an item:
+ * - Attaching an ItemView to RecyclerView
+ * - Detaching an ItemView (scrolling out of viewport) from RecyclerView
+ * - onBindViewHolder
+ * - ItemView layout
+ *
+ * It does *not* measure any of the following work:
+ * - onCreateViewHolder
+ * - RenderThread and GPU Rendering work
+ */
+@LargeTest
+@RunWith(JUnit4::class)
+class RecyclerViewBenchmark {
+
+    @get:Rule
+    val benchmarkRule = BenchmarkRule()
+
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
+
+
+    @Before
+    fun setup() {
+        activityRule.scenario.onActivity {
+            // Initialize the Adapter with fake data.
+            // (Submit null first so both are synchronous for simplicity)
+            it.adapter.submitList(null)
+            it.adapter.submitList(List(100000) { "List Item $it" })
```

Comment:
Genuine question as I'm not sure myself. Does this wait for the views to finish inflating? Does that matter wrt to benchmarking single item view inflation below?

Reply:
Clarified with comments.